security/RFP.md

```diff
@@ -0,0 +1,254 @@
+
+Out os scope:
+1. Hydra head protocol implementation is immune to API attacks -- out of scope because trusted
+
+FIXME: split the RFP or give the option to respond partially. Ask for a detailed estimation of each of the tasks so that we may choose to not ask for everything.
+FIXME: we want the respondant to explain what kind of analysis they will use for each of the assessments.
+
+### Coordinated Hydra Head V1 specification is consistent with the original publication
+
+The Hydra Head v1 specification describes Coordinated Hydra Head V1 Protocol.
+
+This specification provides several important security proofs, especially that the following four properties hold:
+
+* Consistency: No two uncorrupted parties see conflicting transactions confirmed.
+* Liveness: If all parties remain uncorrupted and the adversary delivers all messages, then every transaction becomes confirmed at some point.
+* Soundness: The final UTxO set accepted on the mainchain results from a set of seen transactions.
+* Completeness: All transactions observed as confirmed by an honest party at the end of the protocol are considered on the mainchain.
+FIXME: in the above properties, which one ensures that we don't lock funds in the head?
+
+You will review this specification to give us comments and assess that the above properties hold. The outcome of the review should include, but not being limited to:
+* Identification of any inconsistencies or lack of generality within the specification;
+* Identification of any inconsistencies in the proofs exposed in the specification;
+* Identification of any behavior that could lead, with an adverserial mindset, to one of the above properties to be falsified.
+
+TODO: should we ask, and how, the auditor to provide feedback about the Hydra Head v1 specification on clarity, ambiguity, readability and comprehensibility; it is fit to serve as a foundation for the implementation of the protocol?
+
+FIXME: v1 version of the specification with proof needed.
+
+### Hydra plutus scripts are consistent with Hydra Head v1 specification
+
+The Coordinated Hydra Head V1 specification defines the controls the on-chain validators must implement for a functionning hydra head.
+
+You will review the corresponding artifacts described above to assess that the Hydra plutus scripts are consistent with the Coordinated Hydra Head V1 specification. The outcome of the review should include, bo not being limited to:
+* a review and comment of the mutation based tests applied to the Hydra plutus scripts and, in particular, any adversarial situation that would not be covered by them but should be in the context of this audit;
+* a validation that the Hydra plutus scripts validators do check the transaction constraints defined in the Coordinated Hydra Head V1 specification;
+* any comment about practical Cardano smart contract issue, absent form the specification, which would not be handled by the Hydra plutus scripts;
```

Comment:
It think this overlaps a lot with the next task. Shall we merge them?

Reply:
I don't see both tasks as overlapping and don't feel like we should merge them.

The next task, _ Hydra plutus scripts are immune to common Cardano smart-contract weaknesses_, looks to me quite generic and could apply to any plutus scripts.

When this task, _ Hydra plutus scripts are consistent with Hydra Head v1 specification_ is really about understanding the specification and checking that we apply the same validators in our plutus scripts than in the spec V1.

For instance, we can correctly implement the specification in our validator but be subject to some [hard limits](https://plutus--4604.org.readthedocs.build/en/4604/reference/common-weaknesses/hard-limits.html). That's why I think it has value to identify both task and emphasis on _we don't want you to just check we implemented the spec_ but also something like _the validators can actually run on cardano_.